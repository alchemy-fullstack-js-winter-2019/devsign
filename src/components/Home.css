.Home {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 150px 150px 150px;
  padding: 1 em;
  margin: 20px;
}
header {
  grid-column: 1 / span 4;
  background-color: #84B3DA;
}
footer {
  position: absolute;
  bottom: 0;
}
.sidebar {
  background-color: #84B3DA;
}