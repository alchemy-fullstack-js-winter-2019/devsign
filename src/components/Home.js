import React from 'react';

export default function Home() {
  return (
    <>
      <section>
        <figure>
          <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
          <h3>T_on_A</h3>
        </figure>
      </section>
      <section>
        <label>Post a Tweet</label>
        <input type="text" />
        <button>Tweet</button>
        <button>Add An Image</button>
        <button>Add A Gif</button>
        <button>Add Location</button>
      </section>
      <section>
        <ol>
          <li>List of tweets from users you are following</li>
        </ol>
      </section>
      <aside>
        <ul>
          <section>
            <figure>
              <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
            </figure>
            <h3>Jill Hill</h3>
            <p>Todays lab sucks</p>
          </section>
          <section>
            <figure>
              <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
            </figure>
            <h3>Jack Hill</h3>
            <p>Todays lab is AMAZING</p>
          </section>
          <section>
            <figure>
              <img src="https://pbs.twimg.com/media/C8QsNInXUAAyjZQ.jpg" alt="user avatar" />
            </figure>
            <h3>Hilly Jill</h3>
            <p>Todays lab is all uphill</p>
          </section>
        </ul>
      </aside>
    </>
  );
}
