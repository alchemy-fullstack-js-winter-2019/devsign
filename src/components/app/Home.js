import React, { Fragment } from 'react';
import Header from './Header';
import Trending from '../chirps/Trending';
import ChirpsContainer from '../../containers/PopularChirpsContainer';
import MiniProfileContainer from '../../containers/MiniProfileContainer';
import { Div1, Div2, Section } from './HomeStyles';

export default function DesktopHome() {
  return (
    <Fragment>
      <Header />
      <Section>
        <Div1>
          <MiniProfileContainer />
          <Trending />
        </Div1>
        <Div2>
          <ChirpsContainer />
        </Div2>
      </Section>
    </Fragment>
  );
}
