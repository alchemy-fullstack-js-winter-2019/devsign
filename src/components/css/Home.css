.feed {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;

  h2 {
    color: red;
  }
}
