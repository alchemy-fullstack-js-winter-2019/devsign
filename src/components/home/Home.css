
img {
  max-height: 20vh;
  border-radius: 60px
}
ul {
  padding: 0;
}
.profile  {
  display: none;
  img {
    height: 10vh;
  }
  h3 {
    font-size: 1em;
  }
}

.post {
  margin: 2px;
  background-color: #DEEBED;
  button {
    background-color: #D8D8D8;
  }
  input {
    border: 1px solid #D8D8D8;
  }
}

