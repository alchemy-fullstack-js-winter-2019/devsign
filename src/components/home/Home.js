import React from 'react';
import styles from './Home.css';
import Tweets from '../tweets/Tweets';

const popularTweets = [
  { _id: '0001', text: 'My first tweet', user: { handle: 'Tweedy', userPic: 'http://i.pravatar.cc/1000' } },
  { _id: '0002', text: 'I just flew in from Boise', user: { handle: 'Mocking Jay', userPic: 'http://i.pravatar.cc/1000' } },
  { _id: '0003', text: 'A guy is sitting at an airport bar', user: { handle: 'Parrot', userPic: 'http://i.pravatar.cc/1000' } },
  { _id: '0004', text: '...Tower, please call me a fuel truck.', user: { handle: 'Doug', userPic: 'http://i.pravatar.cc/1000' } },
  { _id: '0005', text: 'KLM 242 expect start up in two hours', user: { handle: 'Polly', userPic: 'http://i.pravatar.cc/1000' } }
];

export default function Home() {
  return (
    <main 
      className={styles.Home}
    > 
      <Tweets tweets={popularTweets} />
    </main>
  );
}
