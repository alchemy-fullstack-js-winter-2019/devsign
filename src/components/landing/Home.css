.main {
  ul {
    padding: 0;
  }
  .profile  {
    display: none;
    img {
      height: 10vh;
    }
    h3 {
      font-size: 1em;
    }
    img {
      max-height: 20vh;
      border-radius: 60px
    }
  }
  
  .post {
    margin: 2px;
    background-color: #DEEBED;
    button {
      background-color: #D8D8D8;
    }
    input {
      border: 1px solid #D8D8D8;
    }
  }
  
  @media screen and (min-width 1000px) {
    .profile {
      
    }
  }
}  
