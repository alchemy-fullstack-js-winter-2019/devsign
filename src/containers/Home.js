import { connect } from 'react-redux';
import Home from '../components/home/Home';
import { getTweetsActions } from '../actions/Tweets';
import { getAllTweets } from '../selectors/Tweets';



const mapStateToProps = state => ({
    tweets: getAllTweets(state)
});

const mapDispatchToProps = dispatch => ({
    fetchTweets() {
        dispatch(getTweetsActions());
    }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Home);
