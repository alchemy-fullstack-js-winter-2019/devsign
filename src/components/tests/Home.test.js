import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Home';
import { MemoryRouter } from 'react-router';

describe('Home snapshot', () => {
  it('can match a snapshot', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
