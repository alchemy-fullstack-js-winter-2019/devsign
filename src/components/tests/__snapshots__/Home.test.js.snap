// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches a snapshot 1`] = `
Array [
  <section
    className="sc-bxivhb jpJyTk"
  >
    <header>
      <h1
        className="warble"
      >
        Warble
      </h1>
      <nav
        className="nav"
      >
        <a
          href="/home"
          name="home"
          onClick={[Function]}
        >
          <span>
            Home
          </span>
          <svg
            alt="home link"
            aria-hidden="true"
            className="svg-inline--fa fa-home fa-w-18 home"
            data-icon="home"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <section
          className="sc-htpNat jAdRHI"
        >
          <button
            onClick={[Function]}
            type="submit"
          >
            <img
              alt="logo"
              src="https://res.cloudinary.com/khortsch/image/upload/v1552673467/logo.png"
            />
          </button>
        </section>
        <a
          href="/profile"
          name="profile"
          onClick={[Function]}
        >
          <img
            alt="prof pic link"
            src="https://res.cloudinary.com/khortsch/image/upload/v1552673467/prof-pic.png"
            width="40px"
          />
          <span>
            Profile
          </span>
        </a>
      </nav>
    </header>
  </section>,
  <section
    className="sc-VigVT goHndl"
  >
    <div
      className="sc-gZMcBi jZrjmQ"
    >
      <aside
        className="sc-iwsKbI hOhYWj"
      >
        <img
          alt="background"
          className="background"
          src="https://res.cloudinary.com/khortsch/image/upload/v1552673467/background_8.44.32_PM.jpg"
        />
        <div
          className="main"
        >
          <img
            alt="prof pic"
            className="profpic"
            src=""
          />
          <div
            className="header"
          >
            <h3>
              
            </h3>
            <p>
              @
              
            </p>
          </div>
        </div>
        <table>
          <thead>
            <tr>
              <th>
                Chirps
              </th>
              <th>
                Followers
              </th>
              <th>
                Following
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                593
              </td>
              <td>
                388
              </td>
              <td>
                301
              </td>
            </tr>
          </tbody>
        </table>
      </aside>
      <aside
        className="sc-bZQynM kLxNdX"
      >
        <h3>
          Trending
        </h3>
        <ul
          className="sc-EHOje mQOyW"
        >
          <li
            className="sc-ifAKCX tCKvm"
          >
            #Bro
          </li>
          <li
            className="sc-ifAKCX tCKvm"
          >
            #Blessed
          </li>
          <li
            className="sc-ifAKCX tCKvm"
          >
            #NoFilter
          </li>
          <li
            className="sc-ifAKCX tCKvm"
          >
            #HumbleBrag
          </li>
          <li
            className="sc-ifAKCX tCKvm"
          >
            #JustSaying
          </li>
          <li
            className="sc-ifAKCX tCKvm"
          >
            #NationalWineDay
          </li>
          <li
            className="sc-ifAKCX tCKvm"
          >
            #RipCity
          </li>
          <li
            className="sc-ifAKCX tCKvm"
          >
            #WCW
          </li>
          <li
            className="sc-ifAKCX tCKvm"
          >
            #MCM
          </li>
        </ul>
      </aside>
    </div>
    <div
      className="sc-gqjmRU dhjaUv"
    >
      <main
        className="sc-htoDjs hMztiY"
      >
        <h2>
          Popular Chirps
        </h2>
        <label>
          <input
            name="search"
            onChange={[Function]}
            placeholder="Search"
            type="text"
            value=""
          />
        </label>
        <ul />
      </main>
    </div>
  </section>,
]
`;
